pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/IngestionDelayTracker.java

```diff
@@ -314,6 +314,22 @@ public void markPartitionForVerification(int partitionGroupId) {
     _partitionsMarkedForVerification.put(partitionGroupId, _clock.millis());
   }
 
+  /*
+   * Method to get timestamp used for the ingestion delay for a given partition.
+   *
+   * @param partitionGroupId partition for which we are retrieving the delay
+   *
+   * @return ingestion delay timestamp in milliseconds for the given partition ID.
+   */
+  public long getPartitionIngestionTimeMs(int partitionGroupId) {
+    // Not protected as this will only be invoked when metric is installed which happens after server ready
+    IngestionTimestamps currentMeasure = _partitionToIngestionTimestampsMap.get(partitionGroupId);
+    if (currentMeasure == null) { // Guard just in case we read the metric without initializing it
+      return 0;
```

Comment:
Suggest returning `Long.MIN_VALUE` by default to be aligned with the segment metadata

Reply:
change applied